tar.go

```diff
@@ -151,6 +151,8 @@ func untarFile(tr *tar.Reader, header *tar.Header, destination string) error {
 		return writeNewFile(filepath.Join(destination, header.Name), tr, header.FileInfo().Mode())
 	case tar.TypeSymlink:
 		return writeNewSymbolicLink(filepath.Join(destination, header.Name), header.Linkname)
+	case tar.TypeLink:
+		return writeNewHardLink(filepath.Join(destination, header.Name), filepath.Join(destination, header.Linkname))
```

Comment:
I'm not super familiar with how this should work -- notice that the line above for symbolic links doesn't join the `destination` and `header.Linkname` values like you do here; is it supposed to be different?